src/components/Icon/SbIcon.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <svg
+    :class="classes"
+    :role="role || 'presentation'"
+    :viewBox="icon.viewBox"
+    v-on="$listeners"
+    v-html="icon.path"
+  ></svg>
+</template>
+
+<script>
+import iconPaths from '../../lib/internal-icons'
+import { availableColors } from '../../utils'
+import { iconSizes } from './utils'
+
+export default {
+  name: 'SbIcon',
+  props: {
+    color: {
+      type: String,
+      default: null,
+      validator: (color) => availableColors.indexOf(color) !== -1,
+    },
+    backgroundColor: {
+      type: String,
+      default: null,
+      validator: (color) => availableColors.indexOf(color) !== -1,
+    },
+    name: {
+      type: String,
+      default: null,
+    },
+    role: {
+      type: String,
+      default: 'presentation',
+    },
+    size: {
+      type: String,
+      default: null,
+      validator: (size) => iconSizes.indexOf(size) !== -1,
+    },
+  },
+  computed: {
+    icon() {
+      const iconDeff = iconPaths[this.name]
+
+      if (!iconDeff) {
+        console.warn(`[SbIcon] - this ${this.name} icon does not exists`)
+      }
+
+      return {
+        path: iconDeff ? iconDeff.path : '',
+        viewBox: iconDeff ? iconDeff.viewbox || '0 0 24 24' : '',
+      }
+    },
+    classes() {
+      return [
+        'sb-icon',
+        this.size ? `sb-icon--${this.size}` : null,
+        this.color ? `sb-icon--color-${this.color}` : null,
+        this.backgroundColor ? `sb-icon--bg-${this.backgroundColor}` : null,
+      ]
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+$icon-colors: (
```

Comment:
We tend to keep the `scss` seperated to it's own file

Reply:
Thank you @lisilinhart. This is just for the job application process, so I'm not surprised there are duplicate PRs. I'll move scss out now.

Reply:
@lisilinhart I moved the `scss` to a separate file and imported from there into the `vue` SFC. I saw that some files import the `scss` into index.js and others import it in the SFC, what is the convention for this project?